src/ui/display.c

```diff
@@ -326,9 +348,34 @@ bool prepare_display_output() {
         strcpy(symbol, token->symbol);
         symbol_len = strlen(token->symbol);
     }
-    strcpy(g_amount, symbol);
-    g_amount[symbol_len] = ' ';
-    format_value(output.value, g_amount + symbol_len + 1);
+
+    if (is_authority_output(output.token_data)) {
+        g_is_authority = true;
+        // set g_authority
+        strcpy(g_authority, symbol);
+        g_authority[symbol_len] = ' ';
+        if (is_mint_authority(output.token_data, output.value)) {
+            strcpy(g_authority + symbol_len + 1, "Mint");
+        } else {
+            if (is_melt_authority(output.token_data, output.value)) {
+                strcpy(g_authority + symbol_len + 1, "Melt");
+            } else {
+                // This authority is unknown, so we treat it as invalid
+                PRINTF("[-] Unknown authority received in value %d\n", output.value);
+                explicit_bzero(&G_context, sizeof(G_context));
+                io_send_sw(SW_INVALID_TX);
+                ui_menu_main();
+                return true;
```

Comment:
I think you can add in the method docstring about this return. What's the meaning of `true` and `false` here?

Reply:
Added a docstring for this